File: java_Report/src/main/java/DataStructures/RStack.java
package DataStructures;

public class RStack {
    private String[] stackList;
    private int length;

    //추가
    public void push(String data){
        if(this.length == 0){
            this.stackList = new String[1];
            this.stackList[0] = data;

        }else{
            String[] newStackList = new String[length+1];

            for(int i =0; i<length; i++){
                newStackList[i] = this.stackList[i];
            }
            newStackList[length] = data;

            this.stackList = new String[length+1];
            this.stackList = newStackList;

        }
        this.length++;


    }
    //제거 겸 뽑기
    public String peek(){
        String popData = stackList[length-1];

        String[] newStackList = new String[length-1];

        for (int i = 0; i < length-1;i++){
            newStackList[i] = stackList[i];
        }

        stackList = new String[length-1];
        stackList = newStackList;

        this.length--;

        return popData;
    }


    //획득(조회)
    public String pop(){
        return stackList[length-1];
    }

    //비어있는지 확인
    public boolean empty(){
        if (length == 0){
            return false;
        }else{
            return true;
        }

    }

}

File: kotlin_Report/src/main/kotlin/RStack.kt
package org.example

class RStack {
    private lateinit var stackArray: Array<String?>
    private var length : Int = 0

    public fun push(data : String){
        if (length == 0){
            this.stackArray = arrayOf(data)

        }else{
            val newStackArray = this.stackArray.copyOf(length+1)
            newStackArray[length] = data

            this.stackArray = Array(length+1){""} //공백은 null 취급 되는가?
            newStackArray.copyInto(this.stackArray,0,0,length+1)
            //endIndex인데 옮길 갯수를 써야하는 의문
        }

        length++
    }

    public fun pop(): String{
        var popData : String? = this.stackArray[length-1].toString()
        var newStackArray =  this.stackArray.copyOf(length-1)

        stackArray = Array(length-1){""}
        newStackArray.copyInto(this.stackArray,0,0,length-1)

        length--

        return popData!!
        //null값을 반환할 수는 없으니 !! 붙이는건 알겠다.
    }

    public fun peek(): String?{
        return stackArray[length-1]
    }


    public fun empty(): String{
        if(length == 0){
            return "No data"
        }else{
            return "Has data"
        }

    }






}

